lib/nats.js

```diff
@@ -1709,15 +1713,17 @@ Client.prototype.publish = function (subject, msg, reply, callback) {
  * @return {Number}
  * @api public
  */
-Client.prototype.subscribe = function (subject, opts, callback) {
-  let qgroup, max
-  if (typeof opts === 'function') {
-    callback = opts
-    opts = undefined
-  } else if (opts && typeof opts === 'object') {
-    // FIXME, check exists, error otherwise..
-    qgroup = opts.queue
-    max = opts.max
+Client.prototype.subscribe = function (subject, options, callback) {
+  const args = callbackShifter(3, Array.prototype.slice.call(arguments))
+  subject = args[0]
+  options = args[1] || {}
+  callback = args[2]
+
+  const qgroup = options.queue
```

Comment:
I really don't like node.. what is that? Meaning that above you just say: `options = args[1] || {}`, but here assume that options is not undefined and you can access options.queue (and .max).

Reply:
a better version of `args[1] ? args[1] : {}` - if the `args[1]` is flasy it sets to empty object.